init-tools.sh

```diff
@@ -66,15 +66,15 @@ if [ ! -e $__DOTNET_PATH ]; then
             *)
                 echo "Unsupported OS '$OSName' detected. Downloading linux-x64 tools."
                 OS=Linux
-                __DOTNET_PKG=dotnet-dev-linux-x64
+                __DOTNET_PKG=dotnet-sdk-${__DOTNET_TOOLS_VERSION}-linux-x64
```

Comment:
For consistency we should do the same thing we did in the other init-tools.sh scripts and factor this out a little differently.

Reply:
Although if you didn't do it in the release branches yet then we can do that separately.

Reply:
What's our plan for keeping these scripts in sync?

Reply:
Yeah I only did it so far on master branches of corefx and coreclr, release branches haven't been changed yet. If we want that change in, I can do it in a separate PR.

>What's our plan for keeping these scripts in sync?

AFAIK there is no plan yet. We have tried to make them as short and consice as possible by moving all of the shared logic to the init-tools scripts on buildtools, and have these just as a bootstrap step. We could do what we do in buildtools repo, where we initialize most of the tools using the CLI bootstrap scripts, but even then we would still have to keep some small scripts that will call into those.

Reply:
One way of making these shorter would be to dogfood our `dotnet-install` scripts that we tell our customers to use.  https://github.com/dotnet/cli/blob/master/scripts/obtain/

Then when a new platform/RID comes along, we add it to the one place (the `dotnet-install` scripts) and this wouldn't even have to change.

Reply:
>What's our plan for keeping these scripts in sync?

We don't have an implemented plan yet. The general idea is to have auto-updated scripts where we can submit PR's to update these common scripts.

>One way of making these shorter would be to dogfood our dotnet-install scripts that we tell our customers to use. https://github.com/dotnet/cli/blob/master/scripts/obtain/

We experimented with this approach in BuildTools and we found that it causes some very interesting failure modes which completely block folks if we cannot download the script (I know same could be said about the CLI itself), and there is also security concerns pulling a script to bootstrap. We can definitely think about it some more but we don't have a clear answer for these yet.

Reply:
One quick idea on a possible approach: we might use a git submodule with all the stuff that we want to share between our repos. Besides the init-tools.sh, it could also contain platform detection code that our build.sh scripts have.

Reply:
Yes that is an idea but taking a dependency on git submodules in all our core repo's has its own set of challenges as well. Now folks  need to know and remember to run git submodule update when syncing.